src/components/cc-orga-member-list/cc-orga-member-list.js

```diff
@@ -0,0 +1,632 @@
+
+/**
+ * @typedef {import('./cc-orga-member-list.types.js').OrgaMemberInviteFormState} OrgaMemberInviteFormState
+ * @typedef {import('./cc-orga-member-list.types.js').OrgaMemberListState} OrgaMemberListState
+ * @typedef {import('./cc-orga-member-list.types.js').Authorisations} Authorisations
+ */
+
+/**
+ * A component showing the list of members belonging to a given organisation.
+ *
+ * The list can be filtered by name or email.
+ * One can also choose to only show users with Two-Factor Auth (2FA) disabled.
+ *
+ * Depending on the current user authorisations:
+ *
+ *  - The current user may remove members,
+ *  - The current user may edit the role of members.
+ *
+ * @cssdisplay block
+ *
+ * @event {CustomEvent<InviteMember>} cc-orga-member-list:invite - Fires the `email` and `role` information inside an object whenever the invite button is clicked.
+ * @event {CustomEvent<DeleteMember>} cc-orga-member-list:leave - Fires the `id` of the member to be removed from the org with their `name` or `email`.
+ */
+
+export class CcOrgaMemberList extends LitElement {
+
+  static get properties () {
+    return {
+      authorisations: { type: Object },
+      inviteMemberForm: { type: Object },
+      members: { type: Object },
+    };
+  }
+
+  static get INIT_INVITE_FORM_STATE () {
+    return {
+      state: 'idle',
+      email: {
+        value: '',
+      },
+      role: {
+        value: 'DEVELOPER',
+      },
+    };
+  };
+
+  static get INIT_AUTHORISATIONS () {
+    return {
+      invite: false,
+      edit: false,
+      delete: false,
+    };
+  };
+
+  constructor () {
+    super();
+
+    /** @type {Authorisations} Sets the authorisations that control the display of the invite form and the edit / delete buttons. */
+    this.authorisations = CcOrgaMemberList.INIT_AUTHORISATIONS;
+
+    /** @type {OrgaMemberInviteFormState} Sets the state of the member invite form. */
+    this.inviteMemberForm = CcOrgaMemberList.INIT_INVITE_FORM_STATE;
+
+    /** @type {OrgaMemberListState} Sets the state of the member list. */
+    this.members = { state: 'loading' };
+
+    this._inviteMemberEmailRef = createRef();
+    this._inviteMemberRoleRef = createRef();
+    this._memberListHeadingRef = createRef();
+    this._noResultMessageRef = createRef();
+
+    new LostFocusController(this, 'cc-orga-member-card', ({ suggestedElement }) => {
+      if (suggestedElement == null && this._noResultMessageRef.value != null) {
+        this._noResultMessageRef.value.focus();
+      }
+      else if (suggestedElement == null) {
+        this._memberListHeadingRef.value.focus();
+      }
+      else {
+        suggestedElement.focusDeleteBtn();
+      }
+    });
+  }
+
+  /**
+   * Check if the member is the last admin of the organisation.
+   *
+   * @param {OrgaMemberState} member - the member to check
+   * @return {boolean} - true if the current user is an admin and there is only one admin left in the organisation
```

Comment:
I prefer `current` over `given` but maybe it's just me.

Reply:
To talk about the connected user I prefer `current` but what you are both pointing out is that this method is in a weird state following my last modifications.
It can be used to check that any member is not the last admin, but it still mentions "current user" in its body / JSDoc.

I'll make it more generic so that it is consistent.

It may be used to check any member so I'll go for "given member" and replace `isCurrentUserAdmin` by `isMemberAdmin` within the method.